tools/runk/src/cgroup.rs

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright 2021 Sony Group Corporation
+//
+
+use anyhow::{anyhow, Result};
+use rustjail::cgroups::fs::Manager as CgroupManager;
+use std::fs;
+use std::path::Path;
+use std::{thread, time};
+
+pub fn destory_cgroup(cgroup_mg: &CgroupManager) -> Result<()> {
+    for path in cgroup_mg.paths.values() {
+        remove_cgroup_dir(Path::new(path))?;
+    }
+
+    Ok(())
+}
+
+fn remove_cgroup_dir(path: &Path) -> Result<()> {
+    let mut retries = 5;
+    let mut delay = time::Duration::from_millis(10);
+    while retries != 0 {
+        if retries != 5 {
+            delay *= 2;
+            thread::sleep(delay);
+        }
+        if remove_dir(path).is_ok() {
+            return Ok(());
+        }
+        retries -= 1;
+    }
+    return Err(anyhow!("Failed to remove paths: {:?}", path));
+}
+
+// Remove cgroup path recursively by removing the subdirectories (sub-cgroups).
+// std::fs::remove_dir_all can not be used to remove the cgroup path because
+// the function also tries to remove files in the directories and fails.
```

Comment:
Thank you for the review.
I removed my`remove_dir` function.
Regarding `sleep`, I use it for a conservative approach to delete cgroups path. If container processes exist in the cgroups, `runk` cannot delete the cgroups directory due to a `Device or resource busy` error. Before calling `remove_cgroup_dir`, `runk` kills the container process, but it may take some time for the process to finish. Hence, `runk` uses `sleep` approach to wait for the process to finish without an infinite loop.
`runc` also takes the same approach:
https://github.com/opencontainers/runc/blob/6ff042023ce8f550a70d3c0bd2588e48df70d939/libcontainer/cgroups/utils.go#L270